src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,163 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Represents a task list
+ */
+public class TaskList {
+
+    private ArrayList<Task> tasks = new ArrayList<>();
+
+    /**
+     * Constructs an empty task list
+     */
+    TaskList() {
+
+    }
+
+    /**
+     * Constructs a task list
+     *
+     * @param data Data containing tasks in the list
+     */
+    TaskList(String data) {
+        if (!data.equals("")) {
+            String[] tasksArray = data.split("\n");
+            for (String task : tasksArray) {
+                this.tasks.add(stringToTask(task));
+            }
+        }
+    }
+
+    private TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    private static Task stringToTask(String input) {
+        String [] taskDetails = input.split(",");
+        String taskDescription = taskDetails[2];
+        boolean isDone;
+        isDone = taskDetails[1].equals("1");
+        if (taskDetails.length == 4) {
```

Comment:
I think it would be better to add a comment specifying the meaning behind hardcoded values (i.e. 4).